From 983639ae3d607893653deaaea834cea37b5df5f9 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian at paral.in>
Date: Fri, 7 Dec 2018 00:20:31 -0800
Subject: [PATCH 2/2] Fix HAVE_INTROSPECTION does not appear in AM_CONDITIONAL

This is a fix for HAVE_INTROSPECTION related errors derived from a fix
originally submitted to the Buildroot project by the author:

Hadrien Boutteville <hadrien.boutteville at gmail.com>

Apply the fix to the same issue in Flatpak:

During autoreconf GOBJECT_INTROSPECTION_CHECK could not be resolved because we
don't have introspection which provides its custom m4 macro. Reconfigure fails
with:

gdk/Makefile.am:196: error: HAVE_INTROSPECTION does not appear in AM_CONDITIONAL
gtk/Makefile.am:1347: error: HAVE_INTROSPECTION does not appear in AM_CONDITIONAL

We avoid to add a copy of introspection.m4 in the m4 directory of libgtk3 by
adding a check, as performed in Systemd.

Signed-off-by: Christian Stewart <christian at paral.in>
---
 configure.ac | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 21f3cf38..2be0d3a2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -316,7 +316,10 @@ if test x$enable_documentation = xyes; then
 fi
 AM_CONDITIONAL(BUILD_DOCUMENTATION, test x$enable_documentation = xyes)
 
-GOBJECT_INTROSPECTION_CHECK([1.40.0])
+m4_ifdef([GOBJECT_INTROSPECTION_CHECK],
+         [GOBJECT_INTROSPECTION_CHECK(1.40.0)],
+         [AM_CONDITIONAL([HAVE_INTROSPECTION], [false])
+          enable_introspection=no])
 
 # gtkdocize greps for ^GTK_DOC_CHECK, so we need to put it on its own line
 m4_ifdef([GTK_DOC_CHECK], [
-- 
2.18.1

